Implemented find command - Closes #109

ppaglilla opened the pull request:
The find command to look for a file or directory (either recursively or not) within the working directory or another specified directory.

I chose to use the java Files class instead of commons FileUtils because the FileUtils aproach is slower (tested in this project: https://github.com/brettryan/io-recurse-tests) and less straightforward.

Please, let me know if you think something should be changed!

ppaglilla committed:
Implemented find command - Closes #109

The find command to look for a file or directory (either recursively or
not) within the working directory or another specified directory.

Sergix commented:
Looks great so far! I really like the layout of the output, and it seems to work fine! 😄 

I only have one suggestion that it immediately noticeable, and that is that the command only matches filenames in the case of an exact match. To make this command more useful, I think it would be a good idea to search for a range of filenames that have a similar name to the input.

For example, say I have a directory in which I am searching for all text files. I should be able to input `find -r .txt` and get a list of results for this command.

I would recommend using the [`String.contains()`](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#contains-java.lang.CharSequence-) function, which should make it pretty simple.

Other than that, I can't think of anything at the moment. Once this is finished, it should be good to go! 👍

ppaglilla commented:
I like the sound of that, it's a great idea! Thanks!
I'll get down to it right away 👍

ppaglilla committed:
Small changes - #109

-Replaced String.equals() with String.contains() so that the output
doesn't return perfect matches only.
-Directories are now added at the beginning of the entries ArrayList.
This way, they are shown first in the output without needing to sort the
list.
-Changed "Search failed" for "Search interrupted" when an IOException is
thrown, as any entries which had already been found will be shown.

Sergix commented:
Thank you so much! Everything looks good to go on my end. :tada:

I'll write the changelog entry for this tomorrow and at that point merge your changes.

If you have any other ideas, feel free to suggest whatever comes to mind.

Also, I would consider joining our Slack channel! If you would like, I can send you an invitation via email. Most discussion happens there, although things have been pretty quiet lately due to busy people with busy lives. 😆

ppaglilla commented:
Terrific! Thank you very much!
I'd love that invitation! I've never used Slack before, but I'll give it a shot.
